### ISSUE TITLE
Split net_buf parsing context from the actual data 

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1683]
**_Reported by Johan Hedberg:_**

To facilitate cheap clones of buffers, we need to split off the data part from the buffer context (meta-data) part. Since we don't want double the complexity of the memory management, the data part should probably be handled with the help of the existing memory pool API in Zephyr. The idea is to follow a similar design as Linux sk_buff, i.e. the data would be ref-counted and able to be referred to from multiple net_buf objects.

One of the requirements for buffer allocation is that it should be possible to happen from ISR context. This means that the memory pool implementation needs some changing to allow this, since right now ISR-based allocation is not possible.

(Imported from Jira ZEP-1835)

**Comment by nashif** [comment]
**_by ethan gao:_**

it's internal implementation optimization and mark it as "TestbyDev", thanks !

**Comment by nashif** [comment]
**_by Tomasz Bursztyka:_**

Taking the task, as discussed at Linaro connect event.

**Comment by nashif** [comment]
**_by Tomasz Bursztyka:_**

It is not exactly blocked by GH-1492, I could get something working with k_mem_slab, which is ISR proof.

**Comment by nashif** [comment]
**_by Tomasz Bursztyka:_**

not the split I did :) I mixed up. I did the net_nbuf split

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Johan Hedberg_ , what's the status of this story?  Is it going to be completed before the merge window for 1.9 closes?

**Comment by nashif** [comment]
**_by Johan Hedberg:_**

I have some code in place, but it's been put aside because of my Bluetooth Mesh work. What's the current estimate of the 1.9 merge window closing?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Johan Hedberg_ , the Feature Merge Window is planned to close in two weeks - by the end of ww32.  See https://github.com/zephyrproject-rtos/zephyr/wiki/Program-Management#planned-trend-and-actual-dates.

**Comment by nashif** [comment]
**_by Andrei Laperie:_**

After discussion with Johan we decided to push this feature to 1.10. This is a nice-to-have feature but requires bit of preparations to become useful, especially if we want to have it for TCP/IP
